src/main/index.ts

```diff
@@ -267,35 +267,32 @@ async function createAppWindows () {
   });
 
   // --- Do not close the IDE (unless exiting the app), only hide it
-  let ideSaved = false;
   ideWindow.on("close", async e => {
-    if (!ideSaved) {
-      // --- Make sure all edited documents are saved
-      e.preventDefault();
-      await sendFromMainToIde({ type: "IdeSaveAllBeforeQuit" });
-      ideSaved = true;
-
-      // --- Try to close the IDE (provided, it's not disposed)
-      ideWindow?.close();
-    } else {
-      // --- The IDE is already saved.
-      if (allowCloseIde) {
-        // --- The emu allows closing the IDE
-        return;
-      }
-
-      // --- Do not allow the IDE close, instead, hide it.
-      e.preventDefault();
-      ideWindow.hide();
-      if (appSettings.windowStates) {
-        // --- Make sure to save the last IDE settings
-        appSettings.windowStates.showIdeOnStartup = false;
-        saveAppSettings();
+    if (allowCloseIde) {
+      // --- The emu allows closing the IDE
+      if (!ideSaved) {
```

Comment:
See how the checks are rearranged. Otherwise we are getting the save-before-quit happen only when the IDE is closed for the very first time and it is not necessarily when the application quits.